settings.go

```diff
@@ -74,6 +80,28 @@ func (s *Settings) Scope(db *gorm.DB, request *goyave.Request, dest interface{})
 		}
 	}
 
+	if !s.DisableSearch && s.SearchOperator != nil && request.Has("search") {
+		query, ok := request.Data["search"].(string)
+		if ok {
+			if len(s.FieldsSearch) == 0 {
+				s.FieldsSearch = s.getSelectableFields(modelIdentity.Columns)
```

Comment:
Don't re-assign to `Settings`. This structure may potentially be used concurrently. Use a temporary variable instead.